common/src/types/v0/store/nexus_persistence.rs

```diff
@@ -43,33 +46,69 @@ pub struct ChildInfo {
 }
 
 /// Key used by the store to uniquely identify a NexusInfo structure.
-pub struct NexusInfoKey(NexusId);
+/// The volume is optional because a nexus can be created which is not associated with a volume.
+pub struct NexusInfoKey {
+    volume_id: Option<VolumeId>,
+    nexus_id: NexusId,
+}
 
-impl From<&NexusId> for NexusInfoKey {
-    fn from(id: &NexusId) -> Self {
-        Self(id.clone())
+impl NexusInfoKey {
+    pub fn new(volume_id: &Option<VolumeId>, nexus_id: &NexusId) -> Self {
+        Self {
+            volume_id: volume_id.clone(),
+            nexus_id: nexus_id.clone(),
+        }
+    }
+}
+
+impl std::fmt::Display for NexusInfoKey {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match &self.volume_id {
+            Some(volume_id) => {
+                write!(f, "volume: {}, nexus: {}", volume_id, self.nexus_id)
+            }
+            None => {
+                write!(f, "nexus: {}", self.nexus_id)
+            }
+        }
     }
 }
 
 impl ObjectKey for NexusInfoKey {
     fn key(&self) -> String {
-        // no key prefix (as it's written by mayastor)
-        self.key_uuid()
+        let namespace = std::env::var("MY_POD_NAMESPACE").unwrap_or_else(|_| "default".into());
+        let nexus_uuid = self.nexus_id.clone();
+        match &self.volume_id {
+            Some(volume_uuid) => {
+                format!(
+                    "/namespace/{}/volume/{}/nexus/{}/info",
```

Comment:
can we keep the namespace logic in a common function that is then used by all types?

Reply:
Done.